### PR TITLE
Changes pfs payment from max fee to requested fee

### LINKED ISSUE #4062: Use PFS requested fee instead of max fee
Currently the client uses the defined `max_fee` when requesting a path from the PFS. This should be changed to use the fee requested by the PFS instead.

https://github.com/raiden-network/raiden/blob/0200586a5c65aa010c1253595f4a8abfe16185ba/raiden/network/pathfinding.py#L401

### PR DESCRIPTION
**Pull request by Dominik1999**
Closes #4062

### CODE REVIEW

File: raiden/tests/unit/test_pfs_integration.py [hunk 1]
```diff
@@ -705,6 +688,7 @@ def query_paths_args(token_network_state, our_address, pfs_max_fee):
         pathfinding_max_fee=pfs_max_fee,
         pathfinding_max_paths=3,
         pathfinding_iou_timeout=500,
+        pathfinding_fee=int(pfs_max_fee / 2),
```
Comment (palango) [inline raiden/tests/unit/test_pfs_integration.py]: Any special reason for setting this to `pfs_max_fee / 2` and not something like `5`?

File: raiden/tests/unit/test_pfs_integration.py [hunk 2]
```diff
@@ -774,37 +758,7 @@ def test_query_paths_with_unrecoverable_pfs_error(query_paths_args):
...
-        expected_requests=2,
-        exception_type=ServiceRequestIOURejected,
-    )
-
-    # second attempt not rejected
-    insufficient_payment[1] = valid_response_json
-    assert_failed_pfs_request(
-        query_paths_args, insufficient_payment, [400, 200], expected_success=True
-    )
+# TODO create tests for sufficient / insufficient payments
```
Comment (palango) [inline raiden/tests/unit/test_pfs_integration.py]: Can you create an issue to track this as well?

Comment (Dominik1999) [reply raiden/tests/unit/test_pfs_integration.py]: https://github.com/raiden-network/raiden/issues/4100

File: raiden/tests/unit/test_pfs_integration.py (hunk 1 above)
Comment (Dominik1999) [reply raiden/tests/unit/test_pfs_integration.py]: I think we can stay in the same range as in `pfs_max_fee` that way, so in the futuer we might test with realistic prices like 10^12 tokens or so

**Pull request merged**